src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java

```diff
@@ -127,7 +128,16 @@ private boolean hasSortField(SearchResultField searchResultField) {
     }
 
     private boolean filterByRole(SearchResultField resultField) {
-        return StringUtils.isEmpty(resultField.getRole()) || userRepository.anyRoleEqualsTo(resultField.getRole());
+        Set<AccessProfile> accessProfiles =  caseDataAccessControl
+            .generateAccessProfilesByCaseTypeId(resultField.getCaseTypeId());
+        Set<String> accessProfileNames = AccessControlService.extractAccessProfileNames(accessProfiles);
+
+        return StringUtils.isEmpty(resultField.getRole())
+            || anyRoleEqualsTo(accessProfileNames, resultField.getRole());
+    }
+
+    private boolean anyRoleEqualsTo(Set<String> accessProfiles, String accessProfile) {
+        return accessProfiles.contains(accessProfile);
```

Comment:
Just the same comments as from `MergeDataToSearchResultOperation`, and same for next class..